src/parse/rules/go_rules.build_defs

```diff
@@ -705,17 +714,34 @@ def _go_library_cmds(complete=True, all_srcs=False, cover=True, filter_srcs=True
     return cmds
 
 
-def _go_binary_cmds(static=False, ldflags='', pkg_config=''):
+def _go_binary_cmds(static=False, ldflags='', pkg_config='', definitions=None):
     """Returns the commands to run for linking a Go binary."""
     _link_cmd = '$TOOLS_GO tool link -tmpdir $TMP_DIR -extld $TOOLS_LD %s -L . -o ${OUT} ' % _GOPATH.replace('-I ', '-L ')
     prefix = _LINK_PKGS_CMD + _go_import_path_cmd(CONFIG.GO_IMPORT_PATH)
 
+    linkerdefs = []
+    if definitions is None:
+        pass
+    elif isinstance(definitions, str):
+        linkerdefs += [f'{definitions}']
+    elif isinstance(definitions, list):
+        linkerdefs += [f'{linkerdef}' for linkerdef in definitions]
+    elif isinstance(definitions, dict):
+        linkerdefs = [k if v is None else f'{k}={v}' for k, v in sorted(definitions.items())]
+
+    defs = ''
+    if len(linkerdefs) > 0:
+        defs += ' '.join([f'-X "{linkerdef}"' for linkerdef in linkerdefs])
```

Comment:
nit: technically it is safe to ignore this conditional and just do

```defs = ' '.join([f'-X "{linkerdef}"' for linkerdef in linkerdefs])```

Reply:
I don't personally like the readability of this, but I also remember this being idiomatic python when I was doing heaps of python work.  Explicit and easy to read from top-to-bottom, left-to-right FTW.  :)  Thanks for the reminder.